app/Image/SizeVariantDefaultFactory.php

```diff
@@ -280,12 +280,19 @@ protected function createSizeVariantInternal(int $sizeVariant, int $maxWidth, in
 			$sv = $this->photo->size_variants->create($sizeVariant, $shortPath, $maxWidth, $maxHeight);
```

Comment:
OK, I found the cause why you made the parameter `filesize` of `SizeVariants::create` optional.

As far as I see it, the only reason why we create the size variant first before we have the actual media file is that we need to get the absolute file path via `$sv->full_path`, because the image handler requires it. Otherwise all attributes (i.e. height, width, etc.) are already known, so we could create the media file first, obtain the file size and _then_ create the size variant. That would lead to a much more consistent approach.

Also note what I said about `->short_path` earlier. As soon as we support anything else than local storage, the image handler must use proper file streams anyway and not rely on absolute paths. From that moment on we could swap the order easily.

However, this is nothing which can be done now easily. So let us keep it that way for the moment.